cypress/integration/e2e/article.bannerPicker.e2e.spec.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-undef */
+/* eslint-disable func-names */
+
+describe('Banner Picker Integration', function () {
+    const visitArticle = (
+        url = 'https://www.theguardian.com/games/2018/aug/23/nier-automata-yoko-taro-interview',
+    ) => {
+        cy.visit(`Article?url=${url}`);
+    };
+
+    const cmpIframe = () => {
+        return cy
+            .get('iframe#sp_message_iframe_208529')
```

Comment:
It seemed to me that the presence of this iframe was a good indicator of the CMP being shown. Does someone more familiar with the CMP know if that's the case? I tried matching on content in the CMP, and while I could see it in the Cypress browser instance, the match was failing (it seems like iframe support is a little tricky in Cypress).

Reply:
Iframes and testing is a nightmare, so I think this is a good work around althought I am not sure the ID `sp_message_iframe_208529` will stay they same?

Maybe add a comment to explain why you used this work around?

Reply:
Yeah great question about the ID. I'll see if I can get someone from commercial dev to take a look.

Reply:
You may have seen it but there's some digging into iframes here for Cypress https://github.com/guardian/dotcom-rendering/blob/main/cypress/integration/e2e/article.elements.spec.js#L5

Reply:
Ah nice. That looks pretty similar to what I'm doing afaict (I was looking at the same blog post reference in the comment). Interesting that chaining onto the returned cypress wrapped element did seem to work when querying for content in that case.

Reply:
@gtrufitt I revisited this and making a small change to match the example you linked to has got things working, so thanks for that! I'm now able to test for content inside the CMP iframe, which I think is a better test (since it's closer to what a user actually sees).